maven/main/java/bsu/fpmi/kwitter/PostCollection.java

```diff
@@ -0,0 +1,161 @@
+package bsu.fpmi.kwitter;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+
+public class PostCollection {
+
+    private ArrayList<Post> collection;
+
+    public PostCollection() {
+        this.collection = new ArrayList<Post>();
+    }
+
+    public PostCollection(ArrayList<Post> collection) {
+        this.collection = collection;
+    }
+
+    public static void fromJson(PostCollection postCollection, String json) {
+        Gson gson = new Gson();
+        postCollection.collection = gson.fromJson(json, new TypeToken<List<Post>>(){}.getType());
+    }
+
+    public static String toJson(PostCollection collection) {
+        Gson gson = new Gson();
+        return gson.toJson(collection);
+    }
+
+    public int size() {
+        return collection.size();
+    }
+
+    public Post get(int index) {
+        return collection.get(index);
+    }
+
+    public boolean match(Post post, Post filterConfig) {
+        if (filterConfig.getId() != -1 && post.getId() != filterConfig.getId()
```

Comment:
The first comparison in each pair seems to be redundant. You hadrly likely will filter by an invalid `filterConfig` (to say, by a post with id = -1), and even if so, you will not catch anything, because matching result will remain `false` due to the `if (filterConfig.getId() != -1...`